docs/guide/key-management.md

```diff
@@ -1,20 +1,170 @@
+
+`basecli keys new` has two inputs (name, password) and two outputs (address, seed).
+
+First, we name our key:
+
+```shelldown
+basecli keys new alice
+```
+
+This will prompt (10 character minimum) password entry which must be re-typed. 
+You'll see:
+
+```
+Enter a passphrase:
+Repeat the passphrase:
+alice		A159C96AE911F68913E715ED889D211C02EC7D70
+**Important** write this seed phrase in a safe place.
+It is the only way to recover your account if you ever forget your password.
+
+pelican amateur empower assist awkward claim brave process cliff save album pigeon intact asset
+```
+
+which shows the address of your key named `alice`, and its recovery seed. We'll use these shortly.
+
+Adding the `--output json` flag to the above command would give this output:
+
+```
+Enter a passphrase:
+Repeat the passphrase:
+{
+  "key": {
+    "name": "alice",
+    "address": "A159C96AE911F68913E715ED889D211C02EC7D70",
+    "pubkey": {
+      "type": "ed25519",
+      "data": "4BF22554B0F0BF2181187E5E5456E3BF3D96DB4C416A91F07F03A9C36F712B77"
+    }
+  },
+  "seed": "pelican amateur empower assist awkward claim brave process cliff save album pigeon intact asset"
+}
+```
+
+To avoid the prompt, it's possible to pipe the password into the command, e.g.:
+
+```
+echo 1234567890 | basecli keys new fred --output json
+```
+
+After trying each of the three ways to create a key, look at them, use:
+
+```
+basecli keys list
+```
+
+to list all the keys:
+
+```
+All keys:
+alice		6FEA9C99E2565B44FCC3C539A293A1378CDA7609
+bob		A159C96AE911F68913E715ED889D211C02EC7D70
+charlie		784D623E0C15DE79043C126FA6449B68311339E5
+```
+
+Again, we can use the `--output json` flag:
 
 ```
+[
+  {
+    "name": "alice",
+    "address": "6FEA9C99E2565B44FCC3C539A293A1378CDA7609",
+    "pubkey": {
+      "type": "ed25519",
+      "data": "878B297F1E863CC30CAD71E04A8B3C23DB71C18F449F39E35B954EDB2276D32D"
+    }
+  },
+  {
+    "name": "bob",
+    "address": "A159C96AE911F68913E715ED889D211C02EC7D70",
+    "pubkey": {
+      "type": "ed25519",
+      "data": "2127CAAB96C08E3042C5B33C8B5A820079AAE8DD50642DCFCC1E8B74821B2BB9"
+    }
+  },
+  {
+    "name": "charlie",
+    "address": "784D623E0C15DE79043C126FA6449B68311339E5",
+    "pubkey": {
+      "type": "ed25519",
+      "data": "4BF22554B0F0BF2181187E5E5456E3BF3D96DB4C416A91F07F03A9C36F712B77"
+    }
+  },
+]
+```
+
+If we want information about a specific key:
+
+```
+basecli keys get charlie --output json
+```
+
+will, for example, return the info for only the "charlie" key returned from the previous `basecoin keys list` command.
+
+Before moving on, let's set some configurations to make `--output json` the default. We can use either the `$BC_HOME` (?) env var or use a `config.toml` that is located in (?)
```

Comment:
BCHOME needs to be mentioned, but it's different.

1) You can set BCHOME to any dir, it looks for a config.toml file in there, and uses `output = "json"` if it is there... It also stores all keys in a subdir of BCHOME (or BC_HOME they are the same)

2) You can `export BC_OUTPUT=json` (or `BCOUTPUT=json`) to make it all json output.

BC_ prefix is standard with viper and seems cleaner to me.  Bucky really liked the eg. TMHOME for tendermint without _ and I made that work here as well with a little setup magic. Pick the one you think is better and use it throughout. (eg. BCHOME, BCOUTPUT, ... or BC_HOME, BC_OUTPUT...)